components/builder-depot/src/server.rs

```diff
@@ -1257,27 +1261,19 @@ fn download_package(req: &mut Request) -> IronResult<Response> {
 
     match route_message::<OriginPackageGet, OriginPackage>(req, &ident_req) {
         Ok(package) => {
-            let temp_path = depot.archive_path(package.get_ident(), &target);
+            let dir = tempdir_in(depot.packages_path()).expect("Unable to create a tempdir!");
+            let file_path = dir.path()
+                .join(depot.archive_name(package.get_ident(), &target));
             let temp_ident = ident.to_owned().into();
-            match s3handler.download(&temp_path, &temp_ident, &target) {
-                Ok(_target) => {
-                    Response::with(status::Ok);
-                    _target
-                }
-                Err(e) => {
-                    warn!("Failed to download package, err={:?}", e);
-                    return Ok(Response::with(status::NotFound));
-                }
-            }
-            if let Some(archive) = depot.archive(package.get_ident(), &target) {
-                match fs::metadata(&archive.path) {
-                    Ok(_) => download_response_for_archive(archive),
-                    Err(_) => Ok(Response::with(status::NotFound)),
-                }
-            } else {
-                // This can happen if the package is not found in the file system for some reason
-                error!("Inconsistentcy between metadata and filesystem!");
-                Ok(Response::with(status::InternalServerError))
+            match s3handler.download(&file_path, &temp_ident, &target) {
+                Ok(archive) => match fs::metadata(&archive.path) {
```

Comment:
This metadata check is also redundant as it is after write_archive in the d/load..